File: my-folder/1039-find-the-town-judge/solution.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        
        if len(trust) < n-1:
            return -1
        counter = {i: 0 for i in range(1, n+1)}
        for t in trust:
            counter[t[0]] -= 1
            counter[t[1]] += 1
        for i in range(1, n+1):
            if counter[i] == n-1:
                return i

        return -1

File: my-folder/1039-find-the-town-judge/solution.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& t) {
       vector<int>res(n+1, 0);
       if(n==1) return 1;

       for(int i=0; i<t.size(); i++) {
           res[t[i][0]]--;
           res[t[i][1]]++;
       }
       for(int i=0; i<=n; i++) {
           if (res[i] == n-1) return i;
       }

       return -1;
    }

};
